src/dataflow-types/src/plan/join/mod.rs

```diff
@@ -40,27 +45,87 @@ use mz_repr::{Datum, Row, RowArena};
 pub use delta_join::DeltaJoinPlan;
 pub use linear_join::LinearJoinPlan;
 
+include!(concat!(env!("OUT_DIR"), "/mz_dataflow_types.plan.join.rs"));
+
 /// A complete enumeration of possible join plans to render.
-#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+#[derive(Arbitrary, Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
 pub enum JoinPlan {
     /// A join implemented by a linear join.
     Linear(LinearJoinPlan),
     /// A join implemented by a delta join.
     Delta(DeltaJoinPlan),
 }
 
+impl TryFrom<ProtoJoinPlan> for JoinPlan {
+    type Error = TryFromProtoError;
+
+    fn try_from(value: ProtoJoinPlan) -> Result<Self, Self::Error> {
+        use proto_join_plan::Kind;
+        let kind = value
+            .kind
+            .ok_or_else(|| TryFromProtoError::MissingField("ProtoJoinPlan::kind".into()))?;
```

Comment:
Hmm there's potential for using `try_into_if_some()` here if we implement `TryFrom<proto_join_plan::Kind> for JoinPlan`

Reply:
Not sure that will make the code more readable. I think we have this pattern already in quite a few places.

Reply:
`TryFromProtoError::missing_field("ProtoJoinPlan::kind")` looks like it will be slightly fewer characters.

Reply:
I change all the constructors to ::missing_field(...) (i've used it in some other places)